format/yaml/src/main/java/org/spongepowered/configurate/yaml/YamlConfigurationLoader.java

```diff
@@ -158,34 +184,97 @@ public Builder nodeStyle(final @Nullable NodeStyle style) {
+         * Get whether comment handling is enabled.
+         *
+         * @return whether comment handling is enabled
+         * @see #commentsEnabled(boolean) for details on comment handling
+         * @since 4.2.0
+         */
+        public boolean commentsEnabled() {
+            return this.enableComments;
+        }
+
+        /**
+         * Set the maximum length of a configuration line.
+         *
+         * <p>The default value is {@code 150}</p>
+         *
+         * @param lineLength the maximum length of a configuration line
+         * @return this builder (for chaining)
+         * @since 4.2.0
+         */
+        public Builder lineLength(final int lineLength) {
+            this.lineLength = lineLength;
+            return this;
+        }
+
+        /**
+         * Get the maximum length of a configuration line.
+         *
+         * @return the maximum length of a configuration line
+         * @see #lineLength(int) for details on the line length
+         * @since 4.2.0
+         */
+        public int lineLength() {
+            return this.lineLength;
+        }
+
         @Override
         public YamlConfigurationLoader build() {
             return new YamlConfigurationLoader(this);
         }
     }
 
+    private final ThreadLocal<YamlConstructor> constructor;
     private final ThreadLocal<Yaml> yaml;
 
     private YamlConfigurationLoader(final Builder builder) {
         super(builder, new CommentHandler[] {CommentHandlers.HASH});
         final LoaderOptions loaderOpts = new LoaderOptions()
             .setAcceptTabs(true)
-            .setProcessComments(false);
+            .setProcessComments(builder.commentsEnabled());
         loaderOpts.setCodePointLimit(Integer.MAX_VALUE);
 
         final DumperOptions opts = builder.options;
         opts.setDefaultFlowStyle(NodeStyle.asSnakeYaml(builder.style));
-        this.yaml = ThreadLocal.withInitial(() -> new Yaml(new Constructor(loaderOpts), new Representer(opts), opts, loaderOpts));
+        opts.setProcessComments(builder.commentsEnabled());
+        opts.setWidth(builder.lineLength());
+        opts.setIndicatorIndent(builder.indent());
+        opts.setIndentWithIndicator(true);
+        // the constructor needs ConfigurationOptions, which is only available when called (loadInternal)
+        this.constructor = ThreadLocal.withInitial(() -> new YamlConstructor(loaderOpts));
+        this.yaml = ThreadLocal.withInitial(() -> new Yaml(this.constructor.get(), new YamlRepresenter(true, opts), opts, loaderOpts));
     }
 
     @Override
     protected void loadInternal(final CommentedConfigurationNode node, final BufferedReader reader) {
-        node.raw(this.yaml.get().load(reader));
+        // the constructor needs ConfigurationOptions for the to be created nodes
+        // and since it's a thread-local, this won't cause any issues
+        this.constructor.get().options = node.options();
+        node.from(this.yaml.get().load(reader));
```

Comment:
There is a regression here: `load` returns null when the file is empty, which `ConfigurationNode#from` can't receive.